src/NServiceBus.AcceptanceTests/Core/Hosting/When_endpoint_starts.cs

```diff
@@ -0,0 +1,44 @@
+﻿namespace NServiceBus.AcceptanceTests.Core.Hosting
+{
+    using System;
+    using System.IO;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using AcceptanceTesting.Customization;
+    using EndpointTemplates;
+    using NUnit.Framework;
+
+    public class When_endpoint_starts : NServiceBusAcceptanceTest
+    {
+        static string basePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, TestContext.CurrentContext.Test.ID);
+
+        [Test]
+        public async Task Should_emit_config_diagnostics()
+        {
+            await Scenario.Define<Context>()
+                .WithEndpoint<MyEndpoint>()
+                .Done(c => c.EndpointsStarted)
+                .Run();
+
+            var endpointName = Conventions.EndpointNamingConvention(typeof(MyEndpoint));
+            var startupDiagnoticsFileName = $"{endpointName}-configuration.json";
+
+            var pathToFile = Path.Combine(basePath, startupDiagnoticsFileName);
+            Assert.True(File.Exists(pathToFile));
+
+            Console.Out.WriteLine(File.ReadAllText(pathToFile));
```

Comment:
Why is this here? Do we want to see this in the test output? If so, this should be `TestContext.WriteLine` instead.